web/app/forms.py

```diff
@@ -47,3 +54,17 @@ def clean(self):
             except PushbulletError:
                 self.add_error('pushbullet_access_token',
                                'Invalid pushbullet access token.')
+
+        if not data['telegram_chat_id']:
+            data['telegram_chat_id'] = None
+
+        if bot and data['telegram_chat_id']:
+            auth = json.loads(data['telegram_chat_id'])
+            data['telegram_chat_id'] = auth['id']
+            telegram_chat_id = data['telegram_chat_id']
+
+            # When a user logs in, we save their chat id and their secret.
+            # We use their secret to validate web hooks,
+            # so we can query for the user and be sure they're not being spoofed.
+            data['telegram_secret'] = token_hex(12)
```

Comment:
Any reason why we need another secret here? Can we just use the secret token of the printer?

Reply:
A user has a telegram login, not a printer. Can't a user have a bunch of printers? Or zero printers?

I could get rid of this secret if they're guaranteed to have at least one printer, but I don't think that's the case.

Reply:
I assume it is used when we send alert notification about a printer. If I'm right here, we already have printer at that moment and we can just use `secret_token` of that printer. But I could be wrong here since I'm still trying to figure out where `telegram_secret` is used and how it can secure the communication.